tools/issue-labeler/labeler/backfill.go

```diff
@@ -1,95 +1,87 @@
-	Number    uint64
+	Number    int
 	Labels    []string
 	OldLabels []string
 }
 
-type IssueUpdateBody struct {
-	Labels []string `json:"labels"`
-}
+func GetIssues(repository, since string) ([]*github.Issue, error) {
+	client := newGitHubClient()
+	owner, repo, err := splitRepository(repository)
+	if err != nil {
+		return nil, fmt.Errorf("invalid repository format: %w", err)
+	}
 
-func GetIssues(repository, since string) ([]Issue, error) {
-	client := &http.Client{}
-	done := false
-	page := 1
-	var issues []Issue
-	for !done {
-		url := fmt.Sprintf("https://api.github.com/repos/%s/issues?since=%s&per_page=100&page=%d", repository, since, page)
-		req, err := http.NewRequest("GET", url, nil)
-		if err != nil {
-			return nil, fmt.Errorf("creating request: %w", err)
-		}
-		req.Header.Add("Accept", "application/vnd.github+json")
-		req.Header.Add("Authorization", "Bearer "+os.Getenv("GITHUB_TOKEN"))
-		req.Header.Add("X-GitHub-Api-Version", "2022-11-28")
-		resp, err := client.Do(req)
-		if err != nil {
-			return nil, fmt.Errorf("listing issues: %v", err)
-		}
-		defer resp.Body.Close()
-		body, err := io.ReadAll(resp.Body)
+	sinceTime, err := time.Parse(time.RFC3339, since)
+	if err != nil {
+		return nil, fmt.Errorf("invalid since time format: %w", err)
+	}
+
+	opt := &github.IssueListByRepoOptions{
+		Since:     sinceTime,
+		State:     "all",
+		Sort:      "updated",
+		Direction: "desc",
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+		},
+	}
+
+	var allIssues []*github.Issue
+	ctx := context.Background()
+
+	for {
+		issues, resp, err := client.Issues.ListByRepo(ctx, owner, repo, opt)
 		if err != nil {
-			return nil, fmt.Errorf("reading response body: %v", err)
+			return nil, fmt.Errorf("listing issues: %w", err)
 		}
-		var newIssues []Issue
-		json.Unmarshal(body, &newIssues)
-		if len(newIssues) == 0 {
-			var err ErrorResponse
-			json.Unmarshal(body, &err)
-			if err.Message == "Bad credentials" {
-				return nil, errors.New("Error from API: Bad credentials")
+
+		// Convert github.Issue to our Issue type
+		for _, issue := range issues {
+			labels := make([]Label, len(issue.Labels))
+			for i, l := range issue.Labels {
+				labels[i] = Label{Name: *l.Name}
 			}
-			glog.Infof("API returned message: %s", err.Message)
-			done = true
-		} else {
-			issues = append(issues, newIssues...)
-			page++
+
+			allIssues = append(allIssues, issue)
 		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
 	}
-	return issues, nil
+
+	return allIssues, nil
 }
 
-func ComputeIssueUpdates(issues []Issue, regexpLabels []RegexpLabel) []IssueUpdate {
+// ComputeIssueUpdates remains the same as it doesn't interact with GitHub API
+func ComputeIssueUpdates(issues []*github.Issue, regexpLabels []RegexpLabel) []IssueUpdate {
 	var issueUpdates []IssueUpdate
 
 	for _, issue := range issues {
-		if len(issue.PullRequest) > 0 {
+		if issue.PullRequestLinks == nil || issue.PullRequestLinks.GetURL() == "" {
```

Comment:
What would you think about using the IsPullRequest helper instead? https://pkg.go.dev/github.com/google/go-github/v68/github#Issue.IsPullRequest

Reply:
sure, I've moved all this to another PR. 
https://github.com/GoogleCloudPlatform/magic-modules/pull/12701/files